api/test-client.js

```diff
@@ -482,10 +482,9 @@ tap.test('connectAndLogin', t => {
 
   t.test('connect failure', t => {
     const creds = {};
-    jujulib.connectAndLogin(url, creds, options, (err, conn, logout) => {
+    jujulib.connectAndLogin(url, creds, options, (err, result) => {
```

Comment:
Can you add a few tests for the async version plz.

Reply:
Will do in a follow-up